src/utils_format_graphite.c

```diff
@@ -199,15 +279,26 @@ int format_graphite(char *buffer, size_t buffer_size, data_set_t const *ds,
       ds_name = ds->ds[i].name;
 
     /* Copy the identifier to `key' and escape it. */
-    status = gr_format_name(key, sizeof(key), vl, ds_name, prefix, postfix,
-                            escape_char, flags);
-    if (status != 0) {
-      ERROR("format_graphite: error with gr_format_name");
-      sfree(rates);
-      return status;
+    if (flags & GRAPHITE_USE_TAGS) {
+      status = gr_format_name_tagged(key, sizeof(key), vl, ds_name, prefix,
+                                     postfix, escape_char, flags);
+      if (status != 0) {
+        ERROR("format_graphite: error with gr_format_name_tagged");
+        sfree(rates);
+        return status;
+      }
```

Comment:
if you would escape this too then this part wouldn't need to be duplicate:
```
 +      if (status != 0) {
 +        ERROR("format_graphite: error with gr_format_name");
 +        sfree(rates);
 +        return status;
 +      }
 +
 +      escape_graphite_string(key, escape_char);
```

Reply:
Actually I thought that it wasn't needed when everything had gone through `gr_copy_escape_part`, but that isn't the case.  I moved the `escape_graphite_string` outside the if, but kept the error checks since the message is slightly different between the 2